README.md

```diff
@@ -1,43 +1,25 @@
-
-```python
-import sys
-sys.path.append('/path/to/intro_machine_learning_using_python')
-import script as s
-```
-
-Check out `tutorial.md` to see an example of the different functionalities of the script!
-
-You can also run the script directly with the following code in a console:
-
-```sh
-python script.py <example.json>
-```
-
-Or in Jupyter notebook with:
-
-```python
-%run script.py <example.json>
-```
-
-In both cases `example.json` stands for the `filename` argument that the script needs. You can use [the file in this repository](example.json) or a similar file of yours. Find more information on how this script works with:
-
-```sh
-python script.py --help
-```
-
-If you run this script, you become proud of yourself.
+# What I'm planning
+
+- Start from the translation job management script created for the first assignment, but expand on it to make it useful for a translation agency rather than an independent translator.
+- Three class attributes (strings) :
+    - Translator
+    - Revisor
+    - Status
+   
+   -> The default value for "Translator" and "Revisor" is "Internal", meaning that an employee of the translation agency took up the job. If the agency assigned the job to a freelancer, the default value can be changed to their name.
+   
+   -> The default value for "Status" is "Created". The status can then be updated as the project progresses to "In translation", "In revision", "Delivered", "Delayed" or "Cancelled". If possible, the script should only accept these six labels to prevent organisational chaos due to everyone using their own labels.
+- Instance and computed attributes remain the same as in the first assignment (with some edits to add the advice from the first assignment's feedback).
+- Add validation for unexpected input (+ for "Status" labels different from the six authorised labels?)
+- Add methods (?) to call the computed attributes and get a result that's more legible than what this currently generates (for example "22 days" instead of "datetime.timedelta(days=22)")
+- The input will still be read from a list of dictionaries in a separate json-file. Those dictionaries will be described in a separate markdown file.
+
+# What I'd like to add but don't know how
+
+- It would be neat if I could do something with the translation memory and termbase of each project. Maybe add a method that opens them for a preview?
+- It would also be super useful to have a way to align a source and a target text and generate a file that can be added to a translation memory. So, to start from two docx-files (or txt-files), split them into sentences and pair each sentence in the source text with the corresponding sentence in the target text and generate a single xml-file with the paired sentences. Ideally, the context (i.e. the surrounding sentences) should also be considered, but if that's not possible an aligned xml would already be awesome.
```

Comment:
This doesn't sound too complicated, at least if you can have both source and target texts as lists of sentences and each sentence has a corresponding one on the other text with the same index. (Think of `zip()`)

With plain text files (e.g. txt) this will be much easier. With docx files, not so much. I found the following package that opens Word files with Python, you can try it out and see what you get: https://python-docx.readthedocs.io/en/latest/user/documents.html

Then I'd need to see what the aligned xml looks like to get an idea of what the output should be.

Reply:
Thanks! I just checked and csv is also a good format for TM import and it has a much "lighter" structure, so I'm going to work with csv rather than xml. :)

Reply:
Both [nltk](https://www.nltk.org/api/nltk.tokenize.html) and [spacy](https://spacy.io/usage/linguistic-features#sbd) offer sentence segmentizers. Based on the languages you need and what you feel more comfortable with, you could use one or the other.

## Example

Given a running text `text` (as one string), we try to get a list of sentences `sentences`.

**nltk**

```python
from nltk.tokenize import sent_tokenize
sentences = sent_tokenize(text, language = 'english')
```

"english" is actually the default value of `language`, but you can provide others. The only docs I found about which was this: https://stackoverflow.com/questions/15111183/what-languages-are-supported-for-nltk-word-tokenize-and-nltk-pos-tag

**spacy**

```python
import spacy
nlp = spacy.load('en_core_web_sm')
doc = nlp(text)
sentences = doc.sents
```

Here the language is defined in the name of the model given to `spacy.load()`, and can have other values: https://spacy.io/usage/models#languages

Reply:
In any case, you still have to know that given a pair of texts, the first sentence of one corresponds to the first sentence of the other, the second to the second, and so on...